src/docs/sdk/sample-format.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Sample format
+sidebar_order: 20
+---
+
+This document defines the Sample format used by Sentry to ingest profiles. This
+document is meant for Sentry SDK developers and maintainers of the profile
+ingestion pipeline.
+
+This format is meant to be cross platform and allow to represent samples
+collected by a profiler.
+
+A profile should always be associated with a transaction.
+
+## Data Model
+
+### Example
+
+```json
+{
+  "debug_meta": { ... },
+  "device": {
+    "architecture": "arm64e",
+    "is_emulator": true,
+    "locale": "en_US",
+    "manufacturer": "Apple",
+    "model": "iPhone14,8"
+  },
+  "environment": "development",
+  "event_id": "",
+  "measurements": { ... },
+  "os": {
+    "build_number": "",
+    "name": "iOS",
+    "version": "16.3",
+  },
+  "platform": "cocoa",
+  "release": "1.0 (9999)",
+  "runtime": {
+    "name": "",
+    "version": ""
+  },
+  "timestamp": "2023-01-01T00:00:00.000Z",
+  "transaction": {
+    "id": "",
+    "name": "",
+    "trace_id": "",
+    "active_thread_id": "",
+  },
+  "version": "1",
+  "profile": { ... },
```

Comment:
Wdyt about adding one example of sample, weight etc here? Would be nice to be able to see the whole format in a single place.

could be like

```
"profile": {
  "samples": [{relative_ts_ns: 0, stack_id: 0, thread_id: 0}...]
},
```
and same for weights and stacks obviously

Reply:
I describe the profiling data separately below https://github.com/getsentry/develop/pull/841/files#diff-f87ed348469343306eb1312b9ba48345f7e7657ba2ae3e62c4d4da89eb0a2b82R199.

I think separating the metadata from the raw data would help to make it easier to read through and reference.

Reply:
Yeah, I understand - I like it when everything in the same place (even if it's not explained) so you minimize any user error from going back/forth between definitions (I find that useful when trying to validate the format). In any case, it's a short doc so not a big concern

Reply:
Alright, I can expand the example to include those fields as well, it doesn't need to be separate.

Also, I'm planning to ship a JSON Schema definition to be able to validate the profile, I think that'd be helpful.

Reply:
➕  for JSON schema. Btw, I do wonder if it would make sense to document the validation constraints or reasons why a "valid" schema may still get dropped, e.g profiles with <= 1 samples.

Reply:
Yes, very good point, that's something I need to add to this document.